### ISSUE TITLE
Create more tree examples

### ISSUE THREAD

**Issue by BlackConure** [TonyGermaneri/canvas-datagrid#187]
### Expected behavior and actual behavior.
The examples in the tutorials do not show well how a tree would be created using Vue.    I have a data source that is returning the full hierarchy of information using 'data' as the child nodes.    The grid does not render any children nodes except for the first one.     

Looking at the documentation, it is not very clear how this is done, but it appears to suggest that data in an expanded node must be populated at runtime only. ??     Is that true, or should it be able to detect child nodes in the data and render the table appropriately? 

### Steps to reproduce the problem.
My data set that I have bound to the canvas-datagrid is structured as:

```
{  id: 1, first: "Fred", last: "Jones", data: [
    {  id: 2, first: "Fred", last: "Jones", data: [...] },
    {  id: 32, first: "Mary", last: "Brown", data: [...] },
    {  id: 49, first: "John", last: "Smith", data: [...] },
    ...
```

Only the first row is rendered.

### Specifications like the version of the project, operating system, or hardware.
Vue: 2.5.17
Vuetify: v 1.4.0
Nuxt: v 1.4.1

**Comment by BlackConure** [comment]
Yeah, let me clarify a bit.   From the documentation it appeared that by adding child data nodes it would render, but in reality, I don't like that approach either.   I prefer the async load as the user expands the node.    This is what I am currently using in Vuetify with v-tree-view.    

***However**,* I am finding that this parent/children node methodology is very problematic as soon as you need to filter and "canvas" the data.    What I am coming to the realization that is needed is a way of displaying data in a hierarchical fashion that allows for infinite scrolling, collapsing and expanding of nodes, as well as filtering and sorting while maintaining the hierarchy.

This is a bit of a challenge because when you filter based on a pattern, you need to retain the lineage of the ancestry of what is found as a result of the search.   By having nodes within nodes, that ability becomes very complex in that it requires all of the data to be loaded as well as navigating the hierarchical structure to find out what is and what is not included in the filter.

It would be better if the data were structured as records where the "depth" is a field of the data and the tree "hierarchy" is constructed relative to depth rather than nesting of children nodes.   The question then becomes, how do you return the lineage with a filtered set of nodes that do not match the filter criteria and how do you create the "canvas" of what to show when some nodes are collapsed and some are expanded?

If you can solve that issue, then you will have a component that can be used for both tree and row based data and still maintain unlimited dataset size with infinite scrolling.

Sorry if this veered off-topic, but your response seemed more about data loading than it did with conforming to Vuetify methodologies.   Maybe change the title?

**Comment by betososa** [comment]
Hi! This is a great project!
I found the talk https://youtu.be/HKd4ODpEUm4

:)
